django/core/paginator.py

```diff
@@ -54,6 +57,16 @@ def page(self, number):
             top = self.count
         return self._get_page(self.object_list[bottom:top], number, self)
 
+    def page_querystring(self, page_number):
+        """
+        Returns a query string for the given page number, preserving any
+        GET parameters present.
+        """
+        params = self.request.GET.copy() \
+            if self.request else QueryDict(mutable=True)
```

Comment:
it's important for the next line after this, where a value is set in that by-default immutable dict.

this is only an issue, when the paginator is used directly and not as part of the generic view. i have added a test.